### ISSUE TITLE
Don't lose stacktrace when error in async `await` statement

### ISSUE THREAD

**Issue by vwkd** [denoland/fresh#966]
Currently, fresh seems to lose the stacktrace for uncaught promise errors by the application, for example in an unguarded `await` statement.

It only traces back to the line `await dev(import.meta.url, "./main.ts");` in the `src/dev.ts` file.

The error looks like this.

```sh
Task start deno run -A --watch=static/,routes/ src/dev.ts
Watcher Process started.
The manifest has been generated for [X] routes and [Y] islands.
error: Uncaught (in promise) Error: Key length is zero
  await import(entrypoint);
  ^
    at async dev (https://deno.land/x/fresh@1.1.0/src/dev/mod.ts:187:3)
    at async file:///[path-to-repo]/src/dev.ts:5:1
Watcher Process finished. Restarting on file change...
```

For example, see https://github.com/denoland/fresh/issues/965

**Comment by lucacasonato** [comment]
cc @bartlomieju. This looks like a bug in Deno related to errors raised in modules being imported from dynamic imports. The stack trace should be the stack of the original error, not of the dynamic import IMO.

**Comment by bartlomieju** [comment]
Fixed in https://github.com/denoland/deno/pull/18664, should be released in Deno v1.32.4 later today.

**Comment by bartlomieju** [comment]
I'm gonna close this now since the bug has been addressed and has been available in Deno for a few patch releases now.
